### PR TITLE
Chapter 4.1

### PR DESCRIPTION
**Pull request by martscht**
Initial commit of Chapter "Introduction to Programming in R". Reviewed and edited by DejanDraschkow.

### CODE REVIEW

**Comment by DejanDraschkow** [comment]
nice! so I am happy with this to be merged :)

File: docs/chapter_ana/references.bib [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+@Misc{Robinson2017a,
+  author = {Robinson, David},
+  title  = {The Impressive Growth of R},
+  month  = {October},
+  year   = {2017},
+  note   = {[Online; accessed Mar 3, 2019]},
+  day    = {10},
+  editor = {Stackoverflow},
+  url    = {https://stackoverflow.blog/2017/10/10/impressive-growth-r/},
+}
```
Comment (jona-sassenhagen) [inline docs/chapter_ana/references.bib:37]: duplicate

File: docs/chapter_ana/references.bib.bak [hunk 2]
```diff
@@ -0,0 +1,59 @@
+% Encoding: UTF-8
```
Comment (jona-sassenhagen) [inline docs/chapter_ana/references.bib.bak]: can this file be deleted

**Comment by jona-sassenhagen** [comment]
Looks good!

File: docs/chapter_ana/references.bib (hunk 1 above)
Comment (martscht) [reply docs/chapter_ana/references.bib:37]: Good catch! Fixed this.

File: docs/chapter_ana/references.bib.bak (hunk 2 above)
Comment (martscht) [reply docs/chapter_ana/references.bib.bak]: Those lines or the entire file?

Comment (jona-sassenhagen) [reply docs/chapter_ana/references.bib.bak]: Is it a backup file? If yes, delete ...

Comment (martscht) [reply docs/chapter_ana/references.bib.bak]: Done.

**Comment by DejanDraschkow** [comment]
should we be merging this? any reason not to?

**Comment by jona-sassenhagen** [comment]
It looks good to me. I can merge, if @martscht is ok with it as is too.

**Comment by martscht** [comment]
To me, it looks good to go

**Pull request merged**